res/qml/WaveformRow.qml

```diff
@@ -0,0 +1,380 @@
+import "." as Skin
+import Mixxx 1.0 as Mixxx
+import QtQuick 2.14
+import QtQuick.Shapes 1.12
+import "Theme"
+
+Item {
+    id: root
+
+    enum MouseStatus {
+        Normal,
+        Bending,
+        Scratching
+    }
+
+    property string group // required
+    property var deckPlayer: Mixxx.PlayerManager.getPlayer(group)
+
+    Item {
+        id: waveformContainer
+
+        property real duration: samplesControl.value / sampleRateControl.value
+
+        anchors.fill: parent
+        clip: true
+
+        Mixxx.ControlProxy {
+            id: samplesControl
+
+            group: root.group
+            key: "track_samples"
+        }
+
+        Mixxx.ControlProxy {
+            id: sampleRateControl
+
+            group: root.group
+            key: "track_samplerate"
+        }
+
+        Mixxx.ControlProxy {
+            id: playPositionControl
+
+            group: root.group
+            key: "playposition"
+        }
+
+        Mixxx.ControlProxy {
+            id: rateRatioControl
+
+            group: root.group
+            key: "rate_ratio"
+        }
+
+        Mixxx.ControlProxy {
+            id: zoomControl
+
+            group: root.group
+            key: "waveform_zoom"
+        }
+
+        Mixxx.ControlProxy {
+            id: introStartPosition
+
+            group: root.group
+            key: "intro_start_position"
+        }
+
+        Mixxx.ControlProxy {
+            id: introEndPosition
+
+            group: root.group
+            key: "intro_end_position"
+        }
+
+        Mixxx.ControlProxy {
+            id: outroStartPosition
+
+            group: root.group
+            key: "outro_start_position"
+        }
+
+        Mixxx.ControlProxy {
+            id: outroEndPosition
+
+            group: root.group
+            key: "outro_end_position"
+        }
+
+        Mixxx.ControlProxy {
+            id: loopStartPosition
+
+            group: root.group
+            key: "loop_start_position"
+        }
+
+        Mixxx.ControlProxy {
+            id: loopEndPosition
+
+            group: root.group
+            key: "loop_end_position"
+        }
+
+        Mixxx.ControlProxy {
+            id: loopEnabled
+
+            group: root.group
+            key: "loop_enabled"
+        }
+
+        Mixxx.ControlProxy {
+            id: mainCuePosition
+
+            group: root.group
+            key: "cue_point"
+        }
+
+        Item {
+            id: waveform
+
+            property real effectiveZoomFactor: rateRatioControl.value * zoomControl.value * 100
```

Comment:
This value is now the inverse what it should be - any reason for reverting what I had put here?

This was the value I had which was ensuring the same behavior than currently:

```suggestion
            property real effectiveZoomFactor: (1 / rateRatioControl.value) * (200 / zoomControl.value)
```

Reply:
Ah thanks. As I said I had to remove the last commits as the PR grew too big and rebasing was no longer possible due to too many conflicts. Fixed in the latest commit.